app/processing/settlement/get-last-settlement.js

```diff
@@ -0,0 +1,31 @@
+const db = require('../../data')
+
+const getLastSettlement = async (settlement, transaction) => {
+  return db.settlement.findOne({
+    transaction,
+    attributes: [
+      'paymentRequestId',
+      'invoiceNumber',
+      'reference',
+      'settled',
+      'settlementDate',
+      'value'
+    ],
+    where: {
+      settlementDate: {
+        [db.Sequelize.Op.lt]: settlement.settlementDate
+      },
+      value: {
+        [db.Sequelize.Op.lt]: settlement.value
+      },
+      invoiceNumber: settlement.invoiceNumber,
```

Comment:
Must be for same payment schedule and same invoice.

Post payment adjustments will need more consideration here as invoice numbers could be split, so we'd probably need to use the prefix rather than the full thing but need to think that through a bit more.